indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -549,3 +618,173 @@ EditableListDisplay.propTypes = {
   management: PropTypes.bool.isRequired,
   canAssignSelf: PropTypes.bool.isRequired,
 };
+
+const filterShape = {
+  key: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string.isRequired,
+      text: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+};
+
+function EditableListFilterSubmenu({filter, addFilter, open, onOpen, isSelectedOption}) {
+  return (
+    <Dropdown
+      scrolling
+      icon={null}
+      className="item"
+      onOpen={onOpen}
+      onBlur={evt => evt.stopPropagation()}
+      open={open}
+      disabled={filter.options.length === 0}
+      trigger={<Dropdown.Item text={filter.text} icon="plus" />}
+    >
+      <Dropdown.Menu>
+        {filter.options.map(option => (
+          <Dropdown.Item
+            key={option.value}
+            value={option.value}
+            text={option.text}
+            disabled={isSelectedOption(option.value)}
+            onClick={(evt, {value: v}) => addFilter(filter, v)}
+          />
+        ))}
+      </Dropdown.Menu>
+    </Dropdown>
+  );
+}
+
+EditableListFilterSubmenu.propTypes = {
+  filter: PropTypes.shape(filterShape).isRequired,
+  addFilter: PropTypes.func.isRequired,
+  open: PropTypes.bool.isRequired,
+  onOpen: PropTypes.func.isRequired,
+  isSelectedOption: PropTypes.func.isRequired,
+};
+
+function EditableListFilter({filterOptions, activeFilters, onChange}) {
+  const [openSubmenu, setOpenSubmenu] = React.useState(-1);
+
+  const sortActiveFilters = filters =>
+    filters.sort((a, b) => {
+      const textCmp = a.filter.text.localeCompare(b.filter.text);
+      if (textCmp === 0 && a.selectedOption && b.selectedOption) {
+        return a.selectedOption.localeCompare(b.selectedOption);
+      }
+      return textCmp;
+    });
+
+  const handleSearchChange = search => onChange({search, filters: activeFilters.filters});
+  const handleFiltersChange = filters => onChange({search: activeFilters.search, filters});
+
+  const addFilter = (filter, selectedOption) => {
+    const newFilter = {
+      key: `${filter.key}_${selectedOption}`,
+      filter,
+      selectedOption,
+    };
+    handleFiltersChange(sortActiveFilters([...activeFilters.filters, newFilter]));
+  };
+
+  const removeFilter = key => {
+    handleFiltersChange(activeFilters.filters.filter(f => f.key !== key));
+  };
+
+  const makeIsSelectedOption = filterKey => option =>
+    activeFilters.filters.some(f => f.key === `${filterKey}_${option}`);
+
+  return (
+    <div>
+      <Dropdown
+        text={Translate.string('Filter')}
+        icon="filter"
+        labeled
+        button
+        className={activeFilters.filters.length > 0 ? 'orange icon' : 'icon'}
+        onClose={() => setOpenSubmenu(-1)}
+      >
+        <Dropdown.Menu>
+          {activeFilters.filters.length > 0 ? (
+            <>
+              <div key="_filters" style={{display: 'flex', flexWrap: 'wrap'}}>
```

Comment:
Weird, I put them there because React was throwing errors but now I can't reproduce them ¯\_(ツ)_/¯

Reply:
That was most likely because it was about a component being created with `.map(...)` or any iteration as React requires them to have keys in order to differentiate them from each other (e.g. added/removed).